File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/radiology/shared/template-style-model.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";

export class TemplateStyleModel {
    public TemplateStyleId: number = 0;
    public HeaderStyle: string = null;
    public TemplateName: string = "";
    public TemplateCode: string = "";
    public FooterStyle: string = null;
    public TemplateId: number = null;
    public CreatedBy: number = null;
    public CreatedOn: string = null;
    public ModifiedBy: number = null;
    public ModifiedOn: string = null;
    public IsActive: boolean = true;
    TemplateStyleFormValidator: FormGroup;
    constructor(
    ) {
        let _formBuilder = new FormBuilder;
        this.TemplateStyleFormValidator = _formBuilder.group({
            TemplateId: [0, Validators.required],
            HeaderStyle: [''],
            FooterStyle: ['']
        })
    }
    public IsDirty(fieldName): boolean {
        if (fieldName == undefined) {
            return this.TemplateStyleFormValidator.dirty;
        }
        else {
            return this.TemplateStyleFormValidator.controls[fieldName].dirty;
        }
    }
    public IsValid(): boolean {
        if (this.TemplateStyleFormValidator.valid) {
            return true;
        }
        else {
            return false;
        }
    }
    public IsValidCheck(fieldName, validator): boolean {
        if (fieldName == undefined) {
            return this.TemplateStyleFormValidator.dirty;
        }
        else {
            return !(this.TemplateStyleFormValidator.hasError(validator, fieldName));
        }
    }
}

File: Code/Components/DanpheEMR.ServerModel/RadiologyModels/TemplateStyleModel.cs
using DanpheEMR.ServerModel.RadiologyModels.DTOs;
using System;
using System.ComponentModel.DataAnnotations;

namespace DanpheEMR.ServerModel.RadiologyModels
{
    public class TemplateStyleModel
    {

        [Key]
        public int TemplateStyleId { get; private set; }

        public string Headerstyle { get; private set; }

        public string FooterStyle { get; private set; }

        public int TemplateId { get; private set; }

        public int CreatedBy { get; private set; }

        public DateTime CreatedOn { get; private set; }

        public int? ModifiedBy { get; private set; }

        public DateTime? ModifiedOn { get; private set; }

        public bool IsActive { get; private set; }
        public static TemplateStyleModel Create(AddTemplateStyleRequest addTemplateStyleRequest, int employeeId)
        {
            return new TemplateStyleModel
            {
                Headerstyle = addTemplateStyleRequest.HeaderStyle,
                FooterStyle = addTemplateStyleRequest.FooterStyle,
                TemplateId = addTemplateStyleRequest.TemplateId,
                CreatedBy = employeeId,
                CreatedOn = DateTime.Now,
                IsActive = true
            };

        }
        public void UpdateStyleContent(string headerStyle, string footerStyle)
        {
            Headerstyle = headerStyle;
            FooterStyle = footerStyle;
        }
        public void UpdateModifiedStatus(int employeeId)
        {
            ModifiedBy = employeeId;
            ModifiedOn = DateTime.Now;
        }   
        public void UpdateIsActiveStatus()
        {
            IsActive = !IsActive;
        }
    }
}
